src/Speech/Operation.php

```diff
@@ -0,0 +1,203 @@
+<?php
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Speech;
+
+use Google\Cloud\Exception\NotFoundException;
+use Google\Cloud\Speech\Connection\ConnectionInterface;
+
+/**
+ * Represents a long-running operation that is the result of a network API call.
+ *
+ * Example:
+ * ```
+ * $operation = $speech->beginRecognizeOperation(
+ *     fopen('audio.flac', 'r')
+ * );
+ * ```
+ */
+class Operation
+{
+    /**
+     * @var ConnectionInterface
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var array
+     */
+    private $info;
+
+    /**
+     * @param ConnectionInterface $connection Represents a connection to the
+     *        Google Cloud Speech API.
+     * @param string $name The operation's name.
+     * @param array $info The operation's data.
+     */
+    public function __construct(ConnectionInterface $connection, $name, array $info = [])
+    {
+        $this->connection = $connection;
+        $this->name = $name;
+        $this->info = $info;
+    }
+
+    /**
+     * Check whether or not the operation is complete. A network request will be
+     * triggered if no cached data exists.
```

Comment:
Definitely! Let's move forward with things as they are now and make a point to follow up on both bigquery/here as a separate task though.